### ISSUE TITLE
What are the dependencies needed for the examples

### ISSUE THREAD

**Issue by stuaxo** [takluyver/terminado#30]
I tried pip installing  

```
tornado
tornado_xstatic
xstatic
```

But it still doesn't seem to be enough, on the website I get a 500 error, python tells me:

```
$ python demos/single.py 
ERROR:tornado.application:Uncaught exception GET / (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:8765', method='GET', uri='/', version='HTTP/1.1', remote_ip='127.0.0.1', headers={'Host': 'localhost:8765', 'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:46.0) Gecko/20100101 Firefox/46.0', 'Accept-Language': 'en-US,en;q=0.5', 'Accept-Encoding': 'gzip, deflate', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'Connection': 'keep-alive'})
Traceback (most recent call last):
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado_xstatic.py", line 36, in _get_xstatic_data_dir
    return cls._cached_xstatic_data_dirs[mod_name]
KeyError: 'termjs'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado/web.py", line 1443, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "demos/single.py", line 14, in get
    ws_url_path="/websocket")
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado/web.py", line 699, in render
    html = self.render_string(template_name, **kwargs)
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado/web.py", line 806, in render_string
    return t.generate(**namespace)
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado/template.py", line 345, in generate
    return execute()
  File "termpage_html.generated.py", line 5, in _tt_execute
    _tt_tmp = xstatic('termjs', 'term.js')  # termpage.html:22
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado_xstatic.py", line 62, in make_url
    version_bit = "?v=" + XStaticFileHandler.get_version({'static_path': ''}, fs_style_path)
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado/web.py", line 2642, in get_version
    abs_path = cls.get_absolute_path(settings['static_path'], path)
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado_xstatic.py", line 49, in get_absolute_path
    root = cls._get_xstatic_data_dir(mod_name)
  File "/home/stu/.virtualenvs/scratch/lib/python3.5/site-packages/tornado_xstatic.py", line 38, in _get_xstatic_data_dir
    xsmod = getattr(__import__('xstatic.pkg', fromlist=[mod_name]), mod_name)
AttributeError: module 'xstatic.pkg' has no attribute 'termjs'
ERROR:tornado.access:500 GET / (127.0.0.1) 6.68ms

```

**Comment by takluyver** [comment]
[XStatic-term.js](https://pypi.python.org/pypi/XStatic-term.js)

PR welcome to document this with a README in the demos directory!

**Comment by stuaxo** [comment]
Awesome, it works :)

I'll do that, should I add some of these deps to the main dir, or are they specific to the demos.


I opened two firefox windows next to each other with pipes running in single.py

http://askubuntu.com/questions/699579/ascii-screensaver-for-the-command-line-or-a-tui  

+ they aren't particularly in sync (one is definitely lagging), also performance of the computer seemed to bog down a bit.

Currently I'm running two gnome terminals with pipes + it seems a lot better.

**Comment by stuaxo** [comment]
Managed to run just one instance of pipes and the performance monitor and had to kill firefox after about 3 minutes.

**Comment by takluyver** [comment]
Tornado is the only dep of terminado itself, IIRC - the others are all specific to the demos.

My instinct is that the slowdown is probably in the JS (term.js), which is separate from terminado. But we should never make claims about performance without measuring stuff ;-)

**Comment by dohnuts** [comment]
On openbsd, i installed tornado and terminado.

```
# python2.7 -v
[lots of impu]
>>> import tornado
import tornado # directory /usr/local/lib/python2.7/site-packages/tornado
# /usr/local/lib/python2.7/site-packages/tornado/__init__.pyc matches /usr/local/lib/python2.7/site-packages/tornado/__init__.py
import tornado # precompiled from /usr/local/lib/python2.7/site-packages/tornado/__init__.pyc
# /usr/local/lib/python2.7/__future__.pyc matches /usr/local/lib/python2.7/__future__.py
import __future__ # precompiled from /usr/local/lib/python2.7/__future__.pyc
>>> import terminado
# terminado.pyc matches terminado.py
import terminado # precompiled from terminado.pyc
# /usr/local/lib/python2.7/site-packages/tornado/web.pyc
[[lots of input]]
packages/tornado/template.pyc
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "terminado.py", line 6, in <module>
    from terminado import SingleTermManager
ImportError: cannot import name SingleTermManager
```

The Symbol is here :
```
$ grep  SingleTermManager /usr/local/lib/python2.7/site-packages/terminado/management.py
class SingleTermManager(TermManagerBase):
        super(SingleTermManager, self).__init__(**kwargs)
```

Whats wrong ?

**Comment by takluyver** [comment]
Looks like you've called a file `terminado.py`, and it's importing that instead of the `terminado` package you've installed.

**Comment by dohnuts** [comment]
I did. Sorry i am not a python user.The rest should be easy .
Fixed
Thanks

**Comment by takluyver** [comment]
No worries, it's an easy mistake to make.

**Comment by takluyver** [comment]
Closed by #31
